src/components/common/ExpressionInput.vue

```diff
@@ -0,0 +1,170 @@
+        return this.currentExpression
+      },
+      set(value) {
+        if (this.expressionIdentifier !== null) {
+          value = {
+            identifier: this.expressionIdentifier,
+            value,
+          }
+        }
+        this.$emit('commitExpressionChange', value)
+      },
+    },
+    expressionContext() {
+      const contactFields = this.subscriberPropertyFields.map(prop => prop.name).concat(defaultContactPropertyFields)
+      const blocks = this.activeFlow.blocks.flatMap(b => [b.name, b.uuid])
+      return {
+        contact: contactFields,
+        flow: blocks,
+        date: defaultDateFields,
+      }
+    },
+    topLevelSuggestions() {
+      return {
+        trigger: '@',
+        values: [
+          {
+            value: '@()',
+            focusText: [-1, -1],
+          },
+          ...Array.from(Object.entries(this.expressionContext)).map(item => `@${item[0]}`),
+        ],
+      }
+    },
+    contextSuggestions() {
+      return Array.from(Object.entries(this.expressionContext)).map((item) => {
+        const name = item[0]
+        return {
+          trigger: `${name}.`,
+          values: item[1].map(val => `${name}.${val}`),
+        }
+      })
+    },
+    methodSuggestions() {
+      return Array.from(this.evaluatorMethods.entries()).map(item => ({
+        trigger: item[0],
+        values: item[1].map(i => ({
+          value: `${i}()`,
+          focusText: [-1, -1],
+        })),
+      }))
+    },
+    evaluatorMethods() {
+      const methods = new Map()
+
+      /* eslint-disable no-restricted-syntax */
+      for (const handler of MethodNodeEvaluatorFactory.defaultHandlers()) {
+        for (const method of handler.handles()) {
+          const trigger = method.substr(0, 2)
+          const upperTrigger = trigger.toUpperCase()
+          const upperMethod = method.toUpperCase()
+          if (methods.has(trigger)) {
+            methods.set(trigger, [...methods.get(trigger), method])
+            methods.set(upperTrigger, [...methods.get(upperTrigger), upperMethod])
+          } else {
+            methods.set(trigger, [method])
+            methods.set(upperTrigger, [upperMethod])
+          }
+        }
+      }
+      /* eslint-enable no-restricted-syntax */
+      return methods
+    },
+    suggestions() {
+      const suggestions = [
+        ...this.contextSuggestions,
+        ...this.methodSuggestions,
+      ]
+      return [
+        ...suggestions,
+        this.topLevelSuggestions,
+      ]
+    }
+  },
+
+  mounted() {
+    const input = this.$refs.input
+    this.suggest = new AutoSuggest({
+      caseSensitive: false,
+      suggestions: this.suggestions,
+      onChange: () => input.dispatchEvent(new Event('input')),
+    }, input)
+
+    this.debounceScroll = debounce(() => { this.didScroll = true }, 100)
+
+    window.addEventListener('scroll', this.debounceScroll)
+  },
+
+  beforeDestroy() {
+    window.removeEventListener('scroll', this.debounceScroll)
```

Comment:
Hi @poojahpatil90 ! It doesn't look like `debouncedScroll()`'s `didScroll` effect is being consumed anywhere. Should we remove references to those altogether?

Reply:
@bzabos You are right!  It was not being used and I have removed it's references.